packages/jest-snapshot/src/getMatchers.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const common = require('./common');
+const fs = require('fs');
+const path = require('path');
+const serializer = require('./serializer');
+const Snapshot = require('./Snapshot');
+
+const testsByPath =  {};
+const getNextIndexFor = (test, filePath) => {
+  const tests = testsByPath[filePath] || (testsByPath[filePath] = {});
+  if (test in tests) {
+    return ++tests[test];
+  }
+  return tests[test] = 0;
+};
+
+module.exports = (filePath, options) => ({
+  toMatchSnapshot: (util, customEquality) => {
+    return {
+      compare(actual, expected) {
+        // Retrieving last test asap to avoid race condition
+        const lastTest = common.getLastTest(filePath);
+
+        const snapshotsPath = path.join(path.dirname(filePath), '__snapshots__');
+        try {
+          const folder = fs.statSync(snapshotsPath);
+          if (!folder.isDirectory()) {
+            fs.mkdirSync(snapshotsPath);
+          }
+        } catch (e) {
+          fs.mkdirSync(snapshotsPath);
+        }
+        // the extension has to be not a jest test one.
+        const snapshotsName = path.join(
+          snapshotsPath,
+          path.basename(filePath) + '.snap'
+        );
+
+
+        if (expected !== undefined) {
+          throw Error('toMatchSnapshot() doesn\' t take parameters (yet)');
```

Comment:
I was thinking of implementing it, so that you can provide your custom key, this will allow cross references through tests, although I'm not sure it's a good idea, cause Jasmine randomize the Spec orders, so you can't rely on something already existing (not that is a real problem as if you're checking for equality it shouldn't matter where you're checking), I want to discuss this further with you 'cause it may actually be a good idea.